lightstep/resource_dashboard.go

```diff
@@ -0,0 +1,218 @@
+				},
+			},
+		},
+	}
+}
+
+func resourceDashboardCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*client.Client)
+	attrs, err := getUnifiedDashboardAttributesFromResource(d)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("failed to get dashboard attributes: %v", err))
+	}
+
+	dashboard := client.UnifiedDashboard{
+		Type:       "dashboard",
+		Attributes: *attrs,
+	}
+
+	created, err := c.CreateUnifiedDashboard(ctx, d.Get("project_name").(string), dashboard)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("failed to create dashboard: %v", err))
+	}
+
+	d.SetId(created.ID)
+	return resourceDashboardRead(ctx, d, m)
+}
+
+func resourceDashboardRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	c := m.(*client.Client)
+
+	dashboard, err := c.GetUnifiedDashboard(ctx, d.Get("project_name").(string), d.Id())
+	if err != nil {
+		apiErr := err.(client.APIResponseCarrier)
+		if apiErr.GetHTTPResponse() != nil && apiErr.GetHTTPResponse().StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return diags
+		}
+		return diag.FromErr(fmt.Errorf("failed to get dashboard: %v", apiErr))
+	}
+
+	if err := setResourceDataFromDashboard(d.Get("project_name").(string), *dashboard, d); err != nil {
+		return diag.FromErr(fmt.Errorf("failed to set dashboard from API response to terraform state: %v", err))
+	}
+
+	return diags
+}
+
+func setResourceDataFromDashboard(project string, dash client.UnifiedDashboard, d *schema.ResourceData) error {
+	if err := d.Set("project_name", project); err != nil {
+		return fmt.Errorf("unable to set project_name resource field: %v", err)
+	}
+
+	if err := d.Set("dashboard_name", dash.Attributes.Name); err != nil {
+		return fmt.Errorf("unable to set dashboard_name resource field: %v", err)
+	}
+
+	if err := d.Set("type", dash.Type); err != nil {
+		return fmt.Errorf("unable to set type resource field: %v", err)
+	}
+
+	var charts []interface{}
+	for _, c := range dash.Attributes.Charts {
+		chart := map[string]interface{}{}
+
+		yMap := map[string]interface{}{}
+
+		if c.YAxis != nil {
+			yMap["max"] = c.YAxis.Max
+			yMap["min"] = c.YAxis.Min
+			chart["y_axis"] = []map[string]interface{}{yMap}
+		}
+
+		chart["query"] = getQueriesFromDashboardResourceData(c.MetricQueries)
+		chart["name"] = c.Title
+		chart["rank"] = c.Rank
+		chart["type"] = c.ChartType
+		chart["id"] = c.ID
+
+		charts = append(charts, chart)
+	}
+
+	if err := d.Set("chart", charts); err != nil {
+		return fmt.Errorf("unable to set chart resource field: %v", err)
+	}
+
+	return nil
+}
+
+func getQueriesFromDashboardResourceData(queriesIn []client.MetricQueryWithAttributes) []interface{} {
+	var queries []interface{}
+	for _, q := range queriesIn {
+		qs := map[string]interface{}{
+			"hidden":       q.Hidden,
+			"display":      q.Display,
+			"query_name":   q.Name,
+			"query_string": q.TQLQuery,
```

Comment:
This file is _mostly_ a clone of `resource_metric_dashboard.go` with the primary change here being the resource data contains _only_ `query_string`.  I thought forking the code would be a cleaner, easier path than refactoring the metric dashboard code with conditionals.

Reply:
outside of [this one](https://github.com/lightstep/terraform-provider-lightstep/pull/98/files#diff-c9ec685f96ccaad128fe473a95c4e4bbea0ba87055d41929e5743cbd59866ac6R558-R561), which other conditionals would we need?

Reply:
Maybe "refactoring with conditionals" was not the right way to express what I thinking. Given the plan is to completely deprecate the lightstep_metric_dashboards resource relatively, it just seemed simpler to fork the code so it's clear what needs to be kept up to data and what needs to be deleted. If the two resources are sharing the same code, coming from the outside, it'd be a little confusing to me what was being used by which resource otherwise.  I don't feel strongly about this as I was mostly just assuming the old code will be deprecated relatively soon.